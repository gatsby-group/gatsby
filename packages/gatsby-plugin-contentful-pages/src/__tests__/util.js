const { createAllQuery, getNodesFor } = require(`../util`)
const PROPS = require(`./fixtures/all-query.json`)
const toJson = val => JSON.stringify(val, null, 2)

describe(`The util module`, () => {
  const contentType = `Product`

  describe(`createAllQuery`, () => {
    it(`should create a GraphQL query for a content type and a field subquery`, () => {
      const subQuery = `
      id
      slug
      tags
      price
      quantity
    `
      const actual = createAllQuery(contentType, subQuery)
      expect(actual).toMatchSnapshot()
    })

    it(`should create a GraphQL query for the entry ID, if no subQuery is passed `, () => {
      const contentType = `myContentType`
      const actual = createAllQuery(contentType)
      expect(actual).toMatchSnapshot()
    })
  })

  describe(`getNodesFor`, () => {
    it(`should retrieve all entries for a content type from the props`, () => {
      const actual = getNodesFor(contentType)(PROPS)
      expect(toJson(actual)).toMatchSnapshot()
    })
  })
})
