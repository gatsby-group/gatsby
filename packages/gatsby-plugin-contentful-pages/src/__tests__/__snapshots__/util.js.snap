// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The util module createAllQuery should create a GraphQL query for a content type and a field subquery 1`] = `
"
  {
    allContentfulProduct {
      edges {
        node {
          
      id
      slug
      tags
      price
      quantity
    
        }
      }
    }
  }
"
`;

exports[`The util module createAllQuery should create a GraphQL query for the entry ID, if no subQuery is passed  1`] = `
"
  {
    allContentfulMyContentType {
      edges {
        node {
          id
        }
      }
    }
  }
"
`;

exports[`The util module getNodesFor should retrieve all entries for a content type from the props 1`] = `
"[
  {
    \\"id\\": \\"c5KsDBWseXY6QegucYAoacS___de\\",
    \\"slug\\": \\"playsam-streamliner-classic-car-espresso\\",
    \\"tags\\": [
      \\"wood\\",
      \\"toy\\",
      \\"car\\",
      \\"sweden\\",
      \\"design\\"
    ],
    \\"price\\": 44,
    \\"quantity\\": 56
  },
  {
    \\"id\\": \\"c3DVqIYj4dOwwcKu6sgqOgg___de\\",
    \\"slug\\": \\"hudson-wall-cup\\",
    \\"tags\\": [
      \\"vase\\",
      \\"flowers\\",
      \\"accessories\\"
    ],
    \\"price\\": 11,
    \\"quantity\\": 101
  },
  {
    \\"id\\": \\"c6dbjWqNd9SqccegcqYq224___de\\",
    \\"slug\\": \\"whisk-beater\\",
    \\"tags\\": [
      \\"kitchen\\",
      \\"accessories\\",
      \\"whisk\\",
      \\"scandinavia\\",
      \\"design\\"
    ],
    \\"price\\": 22,
    \\"quantity\\": 89
  },
  {
    \\"id\\": \\"c4BqrajvA8E6qwgkieoqmqO___de\\",
    \\"slug\\": \\"soso-wall-clock\\",
    \\"tags\\": [
      \\"home décor\\",
      \\"clocks\\",
      \\"interior design\\",
      \\"yellow\\",
      \\"gifts\\"
    ],
    \\"price\\": 120,
    \\"quantity\\": 3
  },
  {
    \\"id\\": \\"c5KsDBWseXY6QegucYAoacS\\",
    \\"slug\\": \\"playsam-streamliner-classic-car-espresso\\",
    \\"tags\\": [
      \\"wood\\",
      \\"toy\\",
      \\"car\\",
      \\"sweden\\",
      \\"design\\"
    ],
    \\"price\\": 44,
    \\"quantity\\": 56
  },
  {
    \\"id\\": \\"c3DVqIYj4dOwwcKu6sgqOgg\\",
    \\"slug\\": \\"hudson-wall-cup\\",
    \\"tags\\": [
      \\"vase\\",
      \\"flowers\\",
      \\"accessories\\"
    ],
    \\"price\\": 11,
    \\"quantity\\": 101
  },
  {
    \\"id\\": \\"c6dbjWqNd9SqccegcqYq224\\",
    \\"slug\\": \\"whisk-beater\\",
    \\"tags\\": [
      \\"kitchen\\",
      \\"accessories\\",
      \\"whisk\\",
      \\"scandinavia\\",
      \\"design\\"
    ],
    \\"price\\": 22,
    \\"quantity\\": 89
  },
  {
    \\"id\\": \\"c4BqrajvA8E6qwgkieoqmqO\\",
    \\"slug\\": \\"soso-wall-clock\\",
    \\"tags\\": [
      \\"home décor\\",
      \\"clocks\\",
      \\"interior design\\",
      \\"yellow\\",
      \\"gifts\\"
    ],
    \\"price\\": 120,
    \\"quantity\\": 3
  }
]"
`;
